### DISCUSSION TITLE
Is there a way to prevent arming/takeoff if in an inappropriate flight mode?

### DISCUSSION THREAD

**Discussion by mirov** [iNavFlight/inav#6417] [Q&A] [answered]
Today I almost had a fly-away on a small INAV FW plane (350g) due to pilot error:  I accidentally armed and took off in position/Altitude hold mode.  Prior to throttle up, and while armed the recorded OSD video shows the flight mode as only "ACRO".  Once it started rolling (I was flying LOS and was not looking at the video real-time) it changes to HOLD.  I've not yet been able to pull the blackbox data, the FC was damaged in the rather abrupt landing that came later.  

My fault clearly, but I'm wondering if there's a way to disallow arming if we're in some inappropriate flight mode (like Position hold).  As I look at the mode configuration page, it seems that you can add "OR" conditions to each mode, but we can't add "AND".  If we could, I'd simply AND my existing arm switch with NOT_POS/ALT_HOLD (if that existed or could be selected of course).  Are there any other modes that come to mind that could result in in a similar situation ?

Looking at the video, the plane did exactly what it was programmed to do:  it took off down the runway and didn't gain any altitude.  As it headed for the edge of the runway/fence I reflexively hit the FS switch (which unfortunately was configured to do a RTH).  It was still bouncing along the ground at this point.   I of course tried to throttle back, but it was already in FS/RTH (auto-throttle is active) and kept going. It climbed very nicely all by itself to the RTH altitude and just moments after it started the turn back I apparently got the RTH cancelled  and with the engine throttled back and the control surface positions not in an ideal state for this situation it went straight in from about 100'.  Amazingly EPP foamies mostly bounce, with an acceptable level of tearing (considering).  As I think about what was going on, it seems that the combination of arming in Pos/Alt Hold and triggering a FS resulted in a fairly nice auto-launch that would have ended with the plane circling (loitering) over the runway had I not aborted things.  Live and learn.

-Russ (new to INAV, but flying R/C since the early 70's)

**Comment by DzikuVx** [comment, marked as answer]
By default arming is disabled when you have poshold enebled. If you disabled `extra_armig_safety` you have to enable it again

**Reply by mirov** [reply]
Perfect !  I'd toggled that to allow no-gps-arming while working in the basement and left it disabled.  Thanks!  -Russ
